src/scripts.js

```diff
@@ -108,27 +120,125 @@ loginSubmitBtn.addEventListener('click', function(e) {
         `;
   });
   displayUserDashboard();
-
   }
 )
 
-function displayUserBookings(userBookings) {
-  
+searchForRoomsBtn.addEventListener('click', function(e) {
+  e.preventDefault();
+  let checkInDate = checkInDateField.value;
+console.log(checkInDate);
+  let newDate = dayjs(checkInDate).format('YYYY/MM/DD')
+
+  let roomTypeToSearch = searchForRoomTypeField.value;
+  console.log(roomTypeToSearch)
+
+  let unavailableRoomByBooking = app.bookingModel.filter(booking => { 
+    if (booking.date === newDate) {
+    return booking;
+    }}
+  )
   
+  let unavailableRoomsRoomNumbers = unavailableRoomByBooking.map(room => {return room.roomNumber});
+
+  unavailableRoomsRoomNumbers = new Set(unavailableRoomsRoomNumbers);
+  unavailableRoomsRoomNumbers = [...unavailableRoomsRoomNumbers];
+
+  let availableRooms = app.roomModel.filter(booking => {
```

Comment:
Having an if statement inside a filter is redundant. It should be written like this instead

`let availableRooms = app.roomModel.filter(booking => {
      return !unavailableRoomsRoomNumbers.includes(booking.number)
})`